src/storage/src/render/persist_sink.rs

```diff
@@ -89,45 +90,81 @@ pub fn render<G>(
                 while let Some((_cap, data)) = input.next() {
                     data.swap(&mut buffer);
 
+                    let mut last_frontier = shared_frontier.borrow().clone();
+                    // TODO: come up with a better default batch size here
```

Comment:
the hint just pre-sizes some internal buffers (think Vec::with_capacity), the size of the chunks is controlled by https://github.com/MaterializeInc/materialize/blob/173a1316a67c73093debcc8892a0dfde7ace7415/src/persist-client/src/lib.rs#L183. if you don't have a good estimate, it's perfectly fine to pass 0. 100 also works

Reply:
We could make the size hint adaptive, by observing what batch sizes we end up writing, dynamically. That's an over-optimization for now, though.

Reply:
@petrosagg and I have some plans on how to improve this impl quite a bit, so I will leave this for now